lib/index.js

```diff
@@ -59,6 +59,7 @@ function AssetGraph(options) {
 
     // this.root might be undefined, in which case urlTools.urlOrFsPathToUrl will use process.cwd()
     this.root = normalizeUrl(urlTools.urlOrFsPathToUrl(this.root, true)); // ensureTrailingSlash
+    this.canonicalRoot = normalizeUrl(urlTools.urlOrFsPathToUrl(this.canonicalRoot, true)); // ensureTrailingSlash
```

Comment:
Hmm, not that I'm aware of. Come to think of it, would we want to support anything but the below?

* `http://example.com`
* `https://example.com`
* `example.com`
* `example.com/foo/bar/`